cmd/main.go

```diff
@@ -357,11 +357,11 @@ func main() {
 		crdUtils = controllers.NewDummyCRDUtils()
 	}
 
-	if Options.EnableDeregisterInactiveGC || Options.EnableDeletedUnregisteredGC {
+	if !Options.EnableKubeAPI && (Options.EnableDeregisterInactiveGC || Options.EnableDeletedUnregisteredGC) {
```

Comment:
`EnableDeletedUnregisteredGC` should probably be enabled even with kube-api, we will need to clear old deleted clusters from the DB

Reply:
@filanov back then we agreed that the deregister-inactive-cluster shouldn't be triggered for kubeapi, however, if CRD & services can be used on the same cluster, we should change this and always enable the worker.
The needed change will be in the [query for the clusters from the DB ](https://github.com/openshift/assisted-service/blob/7088bc266b5953eae1764afdd5f6c373963e36cd/internal/cluster/cluster.go#L1048) not to retrieve clusters that has the kube_key_name and kube_key_namespace (those need to be controlled by the lifecycle of kube_key_namespace/kube_key ClusterDeployment)